library/src/scala/quoted/Expr.scala

```diff
@@ -18,6 +18,14 @@ class Expr[+T] private[scala] {
    */
   final def getValue[U >: T](given qctx: QuoteContext, valueOf: ValueOfExpr[U]): Option[U] = valueOf(this)
 
+  /** Return the value of this expression.
+   *
+   *  Emits an error error and throws if the expression does not contain a value or contains side effects.
+   *  Otherwise returns the value.
+   */
+  final def value[U >: T](given qctx: QuoteContext, valueOf: ValueOfExpr[U]): U =
+    valueOf(this).getOrElse(qctx.throwError(s"Expected a known value. \n\nThe value of: $show\ncould not be recovered using $valueOf", this))
```

Comment:
It seems this method only invites errors in meta-programming, maybe we should just remove it.

Reply:
Without it we would end up writing `expr.getValue.get` all over the place which would crash the macro expansion. This alternative emits an error saying that the call expected a constant value. Pointing to the argument that is not a value.